sentry/src/main/java/io/sentry/ISpan.java

```diff
@@ -32,15 +33,22 @@
   @NotNull
   SentryTraceHeader toSentryTrace();
 
-  /** Sets span timestamp marking this span as finished. */
-  void finish();
+  /**
+   * Sets span timestamp marking this span as finished.
+   *
+   * @return true if transaction has been successfully finished or false if span has been already
+   *     finished
+   */
+  boolean finish();
```

Comment:
wondering why we need to return a boolean here

Reply:
Look into `SentryTracer#finish` - we need to know if finishing the root span ended up in finishing or not.

Reply:
Also a bit surprised about returning this. 
It's also a breaking change.

Reply:
Now moved back to `void`. Technically it's possible that `SentryTracer#finish` will be invoked by two threads, only one will finish the root span and both will capture the same transaction. As far as I can see transactions are de-duplicated on the Sentry backend side, and it's not something that is likely to happen, so I guess it can stay as it is right now.

Reply:
But there's no need to return to the user whether it was captured or not due double calling it though so void works fine